sloth/toolBox.py

```diff
@@ -628,5 +632,119 @@ def mapDataRange_lin(X, y_min=0, y_max=1,
 
     return Y
 
+def intersection_calculations(df_data, corners, area_of_interest, nr_yr, nr_entries, save_dir):
+    """ Calculate spatial mean values for a shapefile in interest.
+
+    This function calculates spatial mean values (example for a specific region or province) taking into account how much (ratio) of the gridpoint is 
```

Comment:
Please insert a line break after 80 characters. 
Even though modern editors can handle longer lines quite well, keeping the line to a maximum of 80 characters makes the code fairly readable on all devices.